src/hierarchical.rs

```diff
@@ -0,0 +1,244 @@
+use std::cmp::{min, max};
+use std::f64::{INFINITY, NEG_INFINITY};
+
+use {Point};
+
+// TODO: Add tests
+// TODO: More efficient `Elements` Iterator
+// TODO: Implement `SLINK` as single-linkage criterion
+// TODO: Implement `CLINK` as alternative complete-linkage criterion
+
+
+/// Hierarchical linkage criteria.
+///
+/// The linkage decides how the distance between clusters is computed.
+///
+/// <script src="https://is.gd/BFouBe"></script>
```

Comment:
This is a little hack to insert `KaTeX` into the rendered documentation, as I described [here](https://gist.github.com/b52/f537e535090f8182dc9a) (shows also the JS source). I would like to include some math but since the discussion around rendered math support has stalled, I wrote this quick fix.